File: typical90/typical90_bf.py
def main():
    num = N
    flag = [-1] * MOD
    for i in range(K):
        if flag[num] > -1:
            loop = i-flag[num]
            idx = flag[num] + (K-i) % loop
            num = flag.index(idx)
            break
        flag[num] = i
        num += sum(map(int, str(num)))
        num %= MOD
    return print(num)


if __name__ == '__main__':
    N, K = map(int, input().split())
    MOD = 10**5

    main()

File: ur_typical90/typical90_bf.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;
using vecll = vector<ll>;

#define rep(i, n) for (ll i = 0; i < ll(n); i++)
#define repitr(id, itr) for (auto& id : itr)
#define all(v) v.begin(), v.end()

ll N, K, MOD = 100000;

int main() {
    cin >> N >> K;

    vecll flag(MOD, -1);
    ll num = N;
    rep(i, K) {
        if (flag[num] > -1) {
            ll loop = i - flag[num];
            ll idx = flag[num] + (K - i) % loop;
            num = find(all(flag), idx) - flag.begin();
            break;
        }
        flag[num] = i;
        repitr(ni, to_string(num)) num += (ni - '0');
        num %= MOD;
    }
    cout << num << '\n';
    return 0;
}
